library/openssl/digest/initialize_spec.rb

```diff
@@ -0,0 +1,141 @@
+require_relative '../../../spec_helper'
+require_relative '../../../library/digest/sha1/shared/constants'
+require_relative '../../../library/digest/sha256/shared/constants'
+require_relative '../../../library/digest/sha384/shared/constants'
+require_relative '../../../library/digest/sha512/shared/constants'
+require 'openssl'
+
+describe "OpenSSL::Digest initialization" do
+  describe "can be called with a digest name" do
+    it "returns a SHA1 object" do
+      OpenSSL::Digest.new("sha1").name.should == "SHA1"
+    end
+
+    it "returns a SHA256 object" do
+      OpenSSL::Digest.new("sha256").name.should == "SHA256"
+    end
+
+    it "returns a SHA384 object" do
+      OpenSSL::Digest.new("sha384").name.should == "SHA384"
+    end
+
+    it "returns a SHA512 object" do
+      OpenSSL::Digest.new("sha512").name.should ==  "SHA512"
+    end
+
+    it "throws an error when called with an unknown digest" do
+      -> { OpenSSL::Digest.new("wd40") }.should raise_error(RuntimeError, /Unsupported digest algorithm \(wd40\)/)
+    end
+
+    it "cannot be called with a symbol" do
+      -> { OpenSSL::Digest.new(:SHA1) }.should raise_error(TypeError, /wrong argument type Symbol/)
+    end
+
+    it "doest not call #to_str on the argument" do
+      name = mock("digest name")
+      name.should_not_receive(:to_str)
+      -> { OpenSSL::Digest.new(name) }.should raise_error(TypeError, /wrong argument type/)
+    end
+  end
+
+  describe "can be called with a digest object" do
+    it "returns a SHA1 object" do
+      OpenSSL::Digest.new(OpenSSL::Digest::SHA1.new).name.should == "SHA1"
+    end
+
+    it "returns a SHA256 object" do
+      OpenSSL::Digest.new(OpenSSL::Digest::SHA256.new).name.should == "SHA256"
+    end
+
+    it "returns a SHA384 object" do
+      OpenSSL::Digest.new(OpenSSL::Digest::SHA384.new).name.should == "SHA384"
+    end
+
+    it "returns a SHA512 object" do
+      OpenSSL::Digest.new(OpenSSL::Digest::SHA512.new).name.should ==  "SHA512"
+    end
+
+    it "cannot be called with a digest class" do
+      -> { OpenSSL::Digest.new(OpenSSL::Digest::SHA1) }.should raise_error(TypeError, /wrong argument type Class/)
+    end
```

Comment:
It seems `"cannot be called with a digest class"` doesn't belong to the `"can be called with a digest object"` section.